Bug 1567912 - Push Fenix nightly AAB to Google Play test track

gbrownmozilla opened the pull request:
### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.






### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1567912

gbrownmozilla commented:
This adds a push-bundle task kind, initially used only for fenix-nightly. 

Provision is made for fenix-beta and fenix-release, but those are disabled for now.

Fenix-nightly push-apk is disabled at the same time.

If the fenix-nightly experience goes smoothly, I hope to enable fenix-beta and fenix-release push-bundle tasks in separate PRs in the near future, then cleanup comments, etc.

There are no immediate plans to push Focus AABs.

gbrownmozilla commented:
`taskgraph target-graph -p taskcluster/test/params/main-repo-cron-nightly.yml --diff` produces:

```
-push-apk-fenix-nightly
+push-bundle-fenix-nightly
```
The beta and release graphs are unchanged.

Unfortunately, `taskgraph target-graph -p taskcluster/test/params/main-repo-cron-nightly.yml --diff -J` produces a complicated diff which conflates these tasks with a third task, comparing `push-apk-fenix-nightly` with `push-apk-focus-nightly` and then `push-apk-focus-nightly` with `push-bundle-fenix-nightly`...but I think it is all correct.

gbrownmozilla commented:
We are not yet ready to enable any Fenix AAB pushes, but I am requesting review now so that we can be ready to proceed when needed.

JohanLorenzo commented on taskcluster/ci/push-bundle/kind.yml:
```diff
@@ -0,0 +1,58 @@
+            with-attributes:
+                build-type:
+                    - fenix-nightly
+                    # coming soon!
```
😃👍

JohanLorenzo commented on taskcluster/ci/push-bundle/kind.yml:
```diff
@@ -0,0 +1,58 @@
+                    # coming soon!
+                    # - fenix-beta
+                    # - fenix-release
+            group-by:
+                attribute: build-type
+            copy-attributes: true
+        worker-type: push-apk
+        run-on-tasks-for: []
+        worker:
+            certificate-alias:
+                by-build-type:
+                    fenix-nightly: fenix-nightly
+                    fenix-beta: fenix-beta
+                    fenix-release: fenix-release
+            commit: true
+            channel:
+                by-build-type:
+                    fenix-nightly: fenix-nightly
+                    fenix-beta: fenix-beta
+                    fenix-release: fenix-release
+            dep:
+                by-level:
+                    '3': false
+                    default: true
+            product: fenix
+        shipping-phase: push
+        treeherder:
+            job-symbol: gp-aab
+            kind: build
+        routes:
+            by-level:
+                '3':
+                  - "notify.email.release-mgmt@mozilla.com.on-failed"
+                'default': []
```
Nit: If I'm not mistaken, this config is almost identical to `push-apk/kind.yml`. I'd recommend we centralize this in a transforms file or maybe use the new `task-context` @bhearsum implemented[1].

That said, I know this file is going to change again before Fenix release is shipped so I won't block on that. 

[1] https://mozilla.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=d657c9ba-ecf8-4c94-a875-b08a0125b9ca&start=420.220708

gbrownmozilla replied:
Thanks. I have thought about that. In the coming weeks, as we test and roll out Fenix AAB, I think it will be less confusing (for me anyway!) to keep separate apk and aab kinds. We can revisit and optimize once the changes have stabilized.

JohanLorenzo reviewed (approved):
Looks good to me modulo one non-blocking nit!

gbrownmozilla commented:
Heh. I guess I got ahead of myself here! 

Before pushing fenix-nightly AABs to the fenix-nightly channel/track, we are going to do additional testing by pushing to a special test track. Therefore, I have removed the changes to push-apk -- we keep pushing nightly APKs as usual - and changed the push-bundle channel to `fenix-aab-nightly`, which pushapkscript will map to the `aab` test track.

Now this change depends on https://github.com/mozilla-releng/scriptworker-scripts/pull/854.
